pubspec.lock

```diff
@@ -14,7 +14,7 @@ packages:
       name: analyzer
       url: "https://pub.dartlang.org"
     source: hosted
-    version: "1.7.2"
+    version: "1.7.1"
```

Comment:
It seems like its downgrading some packages. Maybe run flutter pub upgrade for the latest versions?

Reply:
I did run `flutter pub upgrade` but these packages don't get updated. It seems is because of some dependency constraint.
The result of `flutter pub outdated` is the following:
``` 
Showing outdated packages.
[*] indicates versions that are not the latest available.

Package Name         Current  Upgradable  Resolvable  Latest  

direct dependencies: all up-to-date.

dev_dependencies: all up-to-date.

transitive dependencies:
meta                 *1.3.0   *1.3.0      *1.3.0      1.7.0   

transitive dev_dependencies:
_fe_analyzer_shared  *22.0.0  *22.0.0     *22.0.0     23.0.0  
analyzer             *1.7.1   *1.7.1      *1.7.1      2.0.0   
async                *2.6.1   *2.6.1      *2.6.1      2.7.0   
charcode             *1.2.0   *1.2.0      *1.2.0      1.3.1   
source_gen           *1.0.3   *1.0.3      *1.0.3      1.0.4   
test_api             *0.3.0   *0.3.0      *0.3.0      0.4.2   
You are already using the newest resolvable versions listed in the 'Resolvable' column.
Newer versions, listed in 'Latest', may not be mutually compatible.
```

My flutter version is: 
```
$ flutter --version
Flutter 2.2.3 • channel stable •
https://github.com/flutter/flutter.git
Framework • revision f4abaa0735 (3 settimane fa) • 2021-07-01
12:46:11 -0700
Engine • revision 241c87ad80
Tools • Dart 2.13.4
```